commit d781668386e3adb9c1f171bf2c04396e065189d1 (refs/changes/10/5310/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2019-01-03T23:36:04+00:00 (9 months ago)
    
    joyent/node-zkstream#37 ZKConnectionFSM calculates ping timeout and then doesn't use it
    Reviewed by: Isaac Davis <isaac.davis@joyent.com>

diff --git a/lib/connection-fsm.js b/lib/connection-fsm.js
index b247429..535b040 100644
--- a/lib/connection-fsm.js
+++ b/lib/connection-fsm.js
@@ -412,7 +412,7 @@ ZKConnectionFSM.prototype.ping = function (cb) {
 	var timeout = this.zcf_session.getTimeout() / 8;
 	if (timeout < 2000)
 		timeout = 2000;
-	var timer = setTimeout(onTimeout, this.zcf_session.getTimeout() / 8);
+	var timer = setTimeout(onTimeout, timeout);
 	var t1 = new Date();
 	this.zcf_encoder.write(pkt);
 	function onPacket(pkt2) {
